code/modules/buildmode/submodes/forcemove.dm

```diff
@@ -0,0 +1,52 @@
+// This tool calls forceMove() on an object/mob that is selected with right-click.
+// Once the atom is moved, its selection is cleared to avoid any further moves by accident.
+/datum/buildmode_mode/forcemove
+	key = "forcemove"
+	var/atom/selected_atom		// The atom we want to move
+
+/datum/buildmode_mode/forcemove/show_help(mob/user)
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+	to_chat(user, "<span class='notice'>Left Mouse Button on obj/mob = Select destination</span>")
+	to_chat(user, "<span class='notice'>Right Mouse Button on obj/mob = Select atom to move</span>")
+	to_chat(user, "<span class='notice'><b>Notice:</b> You need to select the movable atom first, then left-click its destination.</span>")
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+
+/datum/buildmode_mode/forcemove/handle_click(user, params, atom/A)
+	var/list/pa = params2list(params)
+	var/left_click = pa.Find("left")
+	var/right_click = pa.Find("right")
+
+	// Selecting the atom to move
+	if(right_click)
+		if(!ismovable(A) || iseffect(A))
+			to_chat(user, "<span class='danger'>You cannot move effects, turfs or areas.</span>")
+			return
+
+		selected_atom = A
+		to_chat(user, "<span class='notice'>'[selected_atom]' is selected to be moved.</span>")
+
+	// Selecting the destination to move to
+	if(left_click)
```

Comment:
make this an early return please

Reply:
Resolved the other two! Not sure what you refer to here - I find it identical to your third point?

Reply:
```dm
if(left_click)
	return
```
then unindent all the code below

Reply:
Im gonna put a bit of input in here. Honestly I think this is better from a readability POV.

Consider the following
```dm
if(left)
    l_stuff()

if(right)
    r_stuff()

if(mid)
    m_stuff()
```

whereas 

```dm
if(left)
    l_stuff()

if(right)
    r_stuff()

if(!mid)
    return

m_stuff()
```

Just looks off.

---

In an ideal world we would have 
```dm
switch(click_type)
    if(LEFT_CLICK)
        l_stuff()

    if(RIGHT_CLICK)
        r_stuff()

    if(MID_CLICK)
        m_stuff()
```
Possible future refactor?

Reply:
Resolved in c5ce5a6